src/Printers/Languages/ZplPrinterCommandSet.ts

```diff
@@ -0,0 +1,142 @@
+import { PrinterCommandLanguage, PrinterOptions } from '../Configuration/PrinterOptions';
+import { PrinterCommandSet, DocumentValidationError } from './PrinterCommandSet';
+import * as Commands from '../../Documents/Commands';
+import { match, P } from 'ts-pattern';
+
+export class ZplPrinterCommandSet extends PrinterCommandSet {
+    private encoder = new TextEncoder();
+
+    get commandLanugage(): PrinterCommandLanguage {
+        return PrinterCommandLanguage.zpl;
+    }
+
+    get documentStartCommand(): Uint8Array {
+        // All ZPL documents start with the start-of-document command.
+        return this.encodeCommand('\n^XA\n');
+    }
+
+    get documentEndCommand(): Uint8Array {
+        // All ZPL documents end with the end-of-document command.
+        return this.encodeCommand('\n^XZ\n');
+    }
+
+    encodeCommand(str: string): Uint8Array {
+        // TODO: ZPL supports omitting the newline, figure out a clever way to
+        // handle situations where newlines are optional to reduce line noise.
+        return this.encoder.encode(str + '\n');
+    }
+
+    transpileCommand(cmd: Commands.IPrinterCommand, outDoc: Commands.CompiledDocument): Uint8Array {
```

Comment:
As a result of the changes in #22 this is a lot closer to how it works now. I generally refer to the new pattern as a _method dispatch lookup table_, where we take a key (in the #22 case either an `enum` or a `symbol` depending on some specific rules) which corresponds to a method to dispatch a command to.

C# has more powerful systems for constructing this table using lookup information, I was trying to find equivalents in TS/JS and came up empty handed.. I initially went with the switch/case-alike pattern because I didn't want to hand-write all of the code I ended up hand-writing in #22. Oh well.

The additional side-effect here is that the transpile operation is generic enough to cram down into the PrinterCommandSet abstract class and leave the Zpl and Epl implementations to _only_ dealing with transpile operations.